src/app/child-dev-project/attendance/add-day-attendance/roll-call/roll-call.component.html

```diff
@@ -1,6 +1,18 @@
 <!-- Individual Student's Page -->
 <div *ngIf="!isFinished() && entries?.length > 0">
-  <app-child-block [entity]="entries[currentIndex].child"></app-child-block>
+  <!--  <app-child-block [entity]="entries[currentIndex].child"></app-child-block>-->
+  <app-child-block
+    [entity]="entries[currentIndex].child"
+    *ngFor="let entry of data?.parts"
+  >
+    <mat-progress-bar
+      mode="determinate"
+      [value]="(entry.currentValue / entry.targetValue) * 100"
```

Comment:
wouldn't it be easier to calculate the value using `currentIndex` and `entries.length`? I dont think it is necessary to calculate all the percentages up-front.